### PR TITLE
fix: Add missing Catalog pageTypes to `@faststore/api`

### PR DESCRIPTION
**Pull request by victorhmp**
## What's the purpose of this pull request?

Add proper support for page types:

- `SubCategory`
- `Collection`
- `Cluster`

This should enable stores to render pages that match those types without issues. Here's an example of one of those issues: 

- https://community.vtex.com/t/problemas-nas-rotas-de-collection-clusters/33353
- https://community.vtex.com/t/enum-storecollectiontype-nao-pode-representar-o-valor-subcategory/33352

## How it works?

Enabling those pageTypes to be returned when the field `collection.type` is queried, instead of throwing an error.

## How to test it?

You can run this branch locally using `yarn develop`, inside `packages/api`.

Here's an example taken from the VTEX Community questions linked above:

**Query for a SubCategory page**

```graphql
{
  collection(slug: "feminino/acessorios/cinto") {
    seo {
      title
      description
    }
    breadcrumbList {
      itemListElement {
        item
        name
        position
      }
    }
    meta {
      selectedFacets {
        key
        value
      }
    }
    id
    type
  }
}
```

**result**

```json
{
  "data": {
    "collection": {
      "seo": {
        "title": "Cinto",
        "description": ""
      },
      "breadcrumbList": {
        "itemListElement": [
          {
            "item": "/feminino",
            "name": "Feminino",
            "position": 1
          },
          {
            "item": "/feminino/acessorios",
            "name": "Acessórios",
            "position": 2
          },
          {
            "item": "/feminino/acessorios/cinto",
            "name": "Cinto",
            "position": 3
          }
        ]
      },
      "meta": {
        "selectedFacets": [
          {
            "key": "category-1",
            "value": "feminino"
          },
          {
            "key": "category-2",
            "value": "acessorios"
          },
          {
            "key": "category-3",
            "value": "cinto"
          }
        ]
      },
      "id": "12",
      "type": "SubCategory"
    }
  }
}
```

### CODE REVIEW

File: packages/api/src/typeDefs/collection.graphql [hunk 1]
```diff
@@ -11,13 +11,21 @@ enum StoreCollectionType {
   """
```
Comment (filipewl) [inline packages/api/src/typeDefs/collection.graphql:11]: Could you include these new ones in this list as well?
https://github.com/vtex/faststore/blob/5102824853b33544cc08897aecb761eb0850c7e2/packages/api/src/typeDefs/collection.graphql#L2

**Review by icazevedo — APPROVED** [review]
LGTM

**Review by filipewl — APPROVED** [review]
Looks good to me as well!

File: packages/api/src/typeDefs/collection.graphql (hunk 1 above)
Comment (victorhmp) [reply packages/api/src/typeDefs/collection.graphql:11]: Sure! Nica catch 😄

**Review by eduardoformiga — APPROVED** [review]

**Pull request merged**